genrl/deep/agents/sac/sac.py

```diff
@@ -266,112 +266,104 @@ def select_action(self, state):
+            min_q_pi = torch.min(q1_pi, q2_pi)
+            policy_loss = ((self.alpha * log_pi) - min_q_pi).mean()
 
-        # gradient step
-        self.q1_optimizer.zero_grad()
-        q1_loss.backward()
-        self.q1_optimizer.step()
+            # gradient step
+            self.q1_optimizer.zero_grad()
+            q1_loss.backward()
+            self.q1_optimizer.step()
 
-        self.q2_optimizer.zero_grad()
-        q2_loss.backward()
-        self.q2_optimizer.step()
+            self.q2_optimizer.zero_grad()
+            q2_loss.backward()
+            self.q2_optimizer.step()
 
-        self.policy_optimizer.zero_grad()
-        policy_loss.backward()
-        self.policy_optimizer.step()
+            self.policy_optimizer.zero_grad()
+            policy_loss.backward()
+            self.policy_optimizer.step()
 
-        # alpha loss
-        if self.entropy_tuning:
-            alpha_loss = -(
-                self.log_alpha * (log_pi + self.target_entropy).detach()
-            ).mean()
+            # alpha loss
+            if self.entropy_tuning:
+                alpha_loss = -(
+                    self.log_alpha * (log_pi + self.target_entropy).detach()
+                ).mean()
 
-            self.alpha_optim.zero_grad()
-            alpha_loss.backward()
-            self.alpha_optim.step()
+                self.alpha_optim.zero_grad()
+                alpha_loss.backward()
+                self.alpha_optim.step()
 
-            self.alpha = self.log_alpha.exp()
-        else:
-            alpha_loss = torch.tensor(0.0).to(self.device)
+                self.alpha = self.log_alpha.exp()
+            else:
+                alpha_loss = torch.tensor(0.0).to(self.device)
 
-        # soft update target params
-        for target_param, param in zip(self.q1_targ.parameters(), self.q1.parameters()):
-            target_param.data.copy_(
-                target_param.data * self.polyak + param.data * (1 - self.polyak)
-            )
+            # soft update target params
+            for target_param, param in zip(
+                self.q1_targ.parameters(), self.q1.parameters()
+            ):
+                target_param.data.copy_(
+                    target_param.data * self.polyak + param.data * (1 - self.polyak)
+                )
 
-        for target_param, param in zip(self.q2_targ.parameters(), self.q2.parameters()):
-            target_param.data.copy_(
-                target_param.data * self.polyak + param.data * (1 - self.polyak)
-            )
+            for target_param, param in zip(
+                self.q2_targ.parameters(), self.q2.parameters()
+            ):
+                target_param.data.copy_(
+                    target_param.data * self.polyak + param.data * (1 - self.polyak)
+                )
 
-        return (q1_loss.item(), q2_loss.item(), policy_loss.item(), alpha_loss.item())
+        # return (q1_loss.item(), q2_loss.item(), policy_loss.item(), alpha_loss.item())
```

Comment:
@Sharad24 Do we need this?

Reply:
Yeah we should either return these from here, or in another function like 'get_logging_params`. The second one makes more sense to me, since we can return more details there.

These will be used in the logging. Would probably be ok to remove for now.

Reply:
There's an issue to keep track of the logging details #156

Reply:
I will do this after I finish cleaning OffPolicyTrainer